modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -145,14 +146,22 @@ object CalibrationsService extends CalibrationObservations {
           }
         } else none.pure[F]
 
-      private def allObservations(pid: Program.Id, selection: ObservationSelection)(using Transaction[F]): F[(List[(Observation.Id, Config.GmosNorth | Config.GmosSouth)])] =
+      private def allValidObservations(obs: Map[Observation.Id, EitherNel[GeneratorParamsService.Error, GeneratorParams]]): List[(Observation.Id, Config.GmosNorth | Config.GmosSouth)] =
+        obs.toList.collect {
+          case (oid, Right(GeneratorParams(Right(_), mode: Config.GmosNorth, _))) => (oid, mode)
+          case (oid, Right(GeneratorParams(Right(_), mode: Config.GmosSouth, _))) => (oid, mode)
+        }
+
+      private def allCalibObservations(obs: Map[Observation.Id, EitherNel[GeneratorParamsService.Error, GeneratorParams]]): List[(Observation.Id, Config.GmosNorth | Config.GmosSouth)] =
+        obs.toList.collect {
+          case (oid, Right(GeneratorParams(_, mode: Config.GmosNorth, _))) => (oid, mode)
+          case (oid, Right(GeneratorParams(_, mode: Config.GmosSouth, _))) => (oid, mode)
+        }
+
+      private def allObservations(pid: Program.Id, selection: ObservationSelection)(using Transaction[F]): F[Map[Observation.Id, EitherNel[GeneratorParamsService.Error, GeneratorParams]]] =
         services
           .generatorParamsService
           .selectAll(pid, selection = selection)
-          .map(_.toList.collect {
-            case (oid, Right(GeneratorParams(_, mode: Config.GmosNorth, _))) => (oid, mode)
-            case (oid, Right(GeneratorParams(_, mode: Config.GmosSouth, _))) => (oid, mode)
-          })
```

Comment:
What if you used the `selection` to determine whether to match on `Right(itcInput)` or not?

Reply:
We also want to filter out science observations without itc